nixos/modules/services/databases/postgresql.nix

```diff
@@ -38,7 +38,7 @@ let
       #     package = pkgs.postgresql_<major>;
       #   };
       # works.
-      base = if cfg.enableJIT then cfg.package.withJIT else cfg.package.withoutJIT;
+      base = if cfg.enableJIT then cfg.package.withJIT else cfg.package;
```

Comment:
This effectively reverts the following valid bugfix: 1682b4cc393dffbadedf8b869a3942965b056743

Reply:
Ok. I see that the contents of my PR is not what we want.

I am still having a hard time to understand the motivation for providing 2 ways to enable Jit instead of just one. As we have seen above, having 2 ways introduces ambiguity. When looking at the module documentation, where can I see that the value specified via `enableJit` overrides the jitiness that I have specified via `package`?

Can you help me find a fix so that I can improve this PR?